### PR TITLE
feat(security): enforce expiry dates for dependency risk exceptions

### LINKED ISSUE #631: security: enforce expiry dates for dependency risk exceptions
## Ziel

Temporäre Dependency-Risk-Ausnahmen sollen automatisch überprüft werden, sobald ihr Ablaufdatum erreicht ist.

## Problem

Das Repo dokumentiert mehrere Dependency-Risiken mit Frist. Ohne automatische Prüfung können solche Ausnahmen dauerhaft liegen bleiben.

## Scope

- Risk-Register maschinenlesbar prüfen.
- CI soll bei abgelaufenen Fristen fehlschlagen.
- Jede Ausnahme braucht nachvollziehbare Metadaten.

## Akzeptanzkriterien

- [ ] Jede Ausnahme enthält: Advisory-ID, Paket, Version, Grund, Owner, Deadline, Blocker und Zielversion.
- [ ] CI prüft Deadlines automatisch.
- [ ] Abgelaufene Deadlines führen zu roter CI.
- [ ] Testfall mit Fake-Eintrag und abgelaufener Deadline vorhanden.
- [ ] Dokumentation erklärt, wann eine Ausnahme zulässig ist.

## Verifikation

```bash
python scripts/check_dependency_risk_register.py
# Erwartung:
# OK: all dependency risk exceptions are valid and not expired
```

## Priorität

P0

## Vorgeschlagener Milestone

`M1 - Release-Gates & CI-Vertrauen`

### PR DESCRIPTION
**Pull request by arn0ld87**
## Summary

- Adds `docs/dependency-risk-exceptions.json` — machine-readable source for all active CVE/Advisory exceptions (3 entries matching the Hardstop-2026-07-30 baseline)
- Adds `backend/scripts/check_dependency_risk_register.py` — stdlib-only checker that validates required fields (advisory_id, package, version_constraint, severity, reason, blocker, target_version, owner, deadline, issue, status) and fails with exit 1 on expired deadlines
- Adds `backend/tests/scripts/test_check_dependency_risk_register.py` — 10 pytest contract tests covering: valid future deadline, expired entry, mixed entries, resolved entries (skipped), missing fields, empty list, missing file, invalid JSON, invalid date format, and boundary condition (deadline == today)
- Adds `schemas/dependency-risk-exceptions.schema.json` — JSON Schema for the exceptions file
- Updates `.github/workflows/cve-monitor.yml` with a new step that runs the checker before pip-audit; expired exceptions fail CI immediately regardless of Hardstop date
- Updates `docs/dependency-risk-register.md` with a reference to the JSON source and a section explaining when an exception is permissible

## Test plan

- [x] `uv run pytest backend/tests/scripts/test_check_dependency_risk_register.py -v` — 10/10 pass
- [x] `python backend/scripts/check_dependency_risk_register.py` — prints OK against real exceptions file
- [ ] Verify CI run on this PR (cve-monitor.yml will run the new step)

Closes #631

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/scripts/check_dependency_risk_register.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    # Deadline-Format prüfen (wenn vorhanden)
+    deadline_raw = entry.get("deadline", "")
+    if deadline_raw and not missing:
+        try:
+            date.fromisoformat(deadline_raw)
+        except ValueError:
+            errors.append(
+                f"Eintrag #{index} ({entry.get('advisory_id', '?')}): "
+                f"Ungültiges Deadline-Format '{deadline_raw}' — erwartet YYYY-MM-DD"
+            )
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/scripts/check_dependency_risk_register.py:113]: Umgesetzt in 82d2a3a: Deadline-Format wird jetzt unabhängig von fehlenden Pflichtfeldern geprüft (Bedingung `not missing` entfernt). Zusätzlich wird `TypeError` neben `ValueError` gefangen, um nicht-String-Werte (z.B. `None` oder Integer) sauber abzufangen.

File: backend/scripts/check_dependency_risk_register.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    """Gibt Fehlermeldung zurück wenn abgelaufen, sonst None."""
+    if entry.get("status") != "open":
+        return None
+    deadline_raw = entry.get("deadline", "")
+    if not deadline_raw:
+        return None
+    try:
+        deadline = date.fromisoformat(deadline_raw)
+    except ValueError:
+        return None  # bereits durch validate_entry gemeldet
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/scripts/check_dependency_risk_register.py:127]: Umgesetzt in 82d2a3a: `check_deadline` fängt jetzt `(ValueError, TypeError)` — bisher nur `ValueError`. Damit crasht das Skript nicht mehr, wenn `deadline_raw` kein String ist.

File: backend/scripts/check_dependency_risk_register.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        default=None,
+        help="Referenzdatum im Format YYYY-MM-DD (default: heute). Für Tests.",
+    )
+    return p.parse_args()
+
+
+def load_exceptions(path: Path) -> list[dict]:
+    if not path.exists():
+        print(f"::error::Datei nicht gefunden: {path}", file=sys.stderr)
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/scripts/check_dependency_risk_register.py]: Umgesetzt in 82d2a3a: `path.exists()` durch `path.is_file()` ersetzt. Fehlermeldung angepasst auf 'Datei nicht gefunden oder kein reguläres File'.

File: backend/scripts/check_dependency_risk_register.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        )
+        sys.exit(1)
+    entries = data["exceptions"]
+    if not isinstance(entries, list):
+        print(
+            "::error::'exceptions' muss eine Liste sein.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    return entries  # type: ignore[return-value]
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/scripts/check_dependency_risk_register.py:92]: Umgesetzt in 82d2a3a: Nach dem `isinstance(entries, list)`-Check wird jetzt `all(isinstance(e, dict) for e in entries)` geprüft. Nicht-dict-Einträge (z.B. Strings) führen zu `::error` + `exit 1`, statt später mit `AttributeError` zu crashen.

File: schemas/dependency-risk-exceptions.schema.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        "issue",
+        "status"
+      ],
+      "additionalProperties": false,
+      "properties": {
+        "advisory_id": {
+          "type": "string",
+          "description": "CVE- oder PYSEC-Bezeichner (z.B. CVE-2026-1839).",
+          "pattern": "^(CVE-\\d{4}-\\d+|PYSEC-\\d{4}-\\d+|GHSA-[a-z0-9-]+)$"
+        },
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply schemas/dependency-risk-exceptions.schema.json:39]: Umgesetzt in 82d2a3a: Pattern von `GHSA-[a-z0-9-]+` auf `GHSA-[A-Za-z0-9-]+` erweitert. Standard-GHSA-IDs (z.B. `GHSA-H756-5GJJ-F8J2`) sind uppercase — das alte Pattern hätte sie fälschlicherweise als ungültig abgelehnt.
